indexd/index/urls/v0.py

```diff
@@ -0,0 +1,115 @@
+        fields (str): comma separated list of fields to return, if not specified return all fields
+        limit (str): max results to return
+        offset (str): where to start the next query from
+    Returns:
+        flask.Response: json list of matching entries
+            `
+                [
+                    {"did": "AAAA-BB", "rev": "1ADs" "urls": ["s3://some-randomly-awesome-url"]},
+                    {"did": "AAAA-CC", "rev": "2Nsf", "urls": ["s3://another-randomly-awesome-url"]}
+                ]
+            `
+    """
+    # parameter validation
+    if kwargs:
+        return Response("Unexpected Parameter(s) {}".format(kwargs), 400)
+    version = version.lower() in ["true", "t", "yes", "y"] if version else None
+    record_list = urls.driver.query_urls(exclude=exclude, include=include,
+                                         only_versioned=version,
+                                         offset=int(offset), limit=int(limit))
+
+    fields = fields or "did,urls"
+    fields_dict = requested_fields(fields)
+
+    results = []
+    for record in record_list:
+        c_response = {}
+        if fields_dict.get("did"):
+            c_response["did"] = record[0]
+        if fields_dict.get("urls"):
+            c_response["urls"] = record[1].split(",") if record[1] else []
+        results.append(c_response)
+
+    response = Response(json.dumps(results, indent=2, separators=(', ', ': ')), 200, mimetype="application/json")
+    return response
+
+
+@urls.route("/metadata/q")
+@request_args_to_params
+def query_metadata(key, value, url=None, version=None, fields=None, limit="100", offset="0", **kwargs):
+    """ Queries indexes by URLs metadata key and value
+    Args:
+        key (str): metadata key
+        value (str): metadata value for key
+        url (str): full url or pattern for limit to
+        fields (str): comma separated list of fields to return, if not specified return all fields
+        version (str): filter only records with a version number
+        limit (str): max results to return
+        offset (str): where to start the next query from
+    """
+
+    if kwargs:
+        return Response("Unexpected Parameter(s) {}".format(kwargs), 400)
+
+    version = version.lower() in ["true", "t", "yes", "y"] if version else None
+    record_list = urls.driver.query_metadata_by_key(key, value, url=url,
+                                                    only_versioned=version,
+                                                    offset=int(offset), limit=int(limit))
+    fields = fields or "did,urls,rev"
+    fields_dict = requested_fields(fields)
+
+    results = []
+    for record in record_list:
+        c_response = {}
+        if fields_dict.get("did"):
+            c_response["did"] = record[0]
+        if fields_dict.get("urls"):
+            c_response["urls"] = [record[1]]
+        if fields_dict.get("rev"):
+            c_response["rev"] = record[2]
+        results.append(c_response)
```

Comment:
if you change the result from `query_metadata_by_key` then this chunk of logic isn't necessary and you can just go straight to returning the JSON